### PR TITLE
Add deprecation notice to readme.txt

### PR DESCRIPTION
**Pull request by JacklynBiggin**
## What

Adds a short deprecation notice to the top of the `readme.txt` file explaining that the plugin is no longer supported. It directs readers to the announcement blockpost about the blocks -> monorepo merge.

## Why

Development in this repo has ceased, and all future contributions should take place in the WooCommerce monorepo.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Read the modified lines in `readme.txt`. Validate they make sense and are correct, both grammatically and logically. Feel free to make changes as necessary. 
2.
3.

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:

* [ ] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:

* [ ] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Add suggested changelog entry here.

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -8,6 +8,11 @@ Stable tag: 11.6.2
 License: GPLv3
 License URI: https://www.gnu.org/licenses/gpl-3.0.html
 
+== This Plugin is Deprecated ==
+As of December 2023, we [merged the standalone WooCommerce Blocks plugin into WooCommerce itself](https://developer.woo.com/2023/12/11/the-woo-blocks-merge-to-the-monorepo-is-complete-along-with-monorepo-tool-enhancements/). All future development will now take place directly in WooCommerce.
+
+If you're building Blocks-based functionality for WooCommerce, you no longer need this plugin.
```
Comment (nerrad) [inline readme.txt]: We can remove this line and include something like this instead:

> Any sites with WooCommerce 8.5.0 or greater will have access to the latest WooCommerce Blocks.

**Review by nerrad — COMMENTED** [review]
PRs can't be merged into this repo because it is archived. But suggestions here can be SVN'd directly to the WordPress.org repository once we're happy with it.

**Comment by nerrad** [comment]
Let's see if this stays open so we can merge the readme.txt change :)

**Comment by danielwrobert** [comment]
> PRs can't be merged into this repo because it is archived.

@nerrad it doesn't seem that this repo has been archived _yet_. I think we can go ahead and merge this PR and then, once we confirm implications, go ahead and archive it.

File: readme.txt (hunk 1 above)
Comment (JacklynBiggin) [reply readme.txt]: Good call. Making that change now!

**Comment by JacklynBiggin** [comment]
@nerrad Changed that line as requested!

**Review by nerrad — APPROVED** [review]

**Pull request merged**